test/unit/deadline_job_attachments/api/test_attachment.py

```diff
@@ -0,0 +1,190 @@
+            )
+
+
+@pytest.mark.parametrize("manifest_case_key", MOCK_MANIFEST_CASE.keys())
+def test_attachment_download_single_to_current(temp_assets_dir, manifest_case_key):
+    with patch.object(api._session, "get_boto3_session") as session_mock:
+        with open(
+            os.path.join(temp_assets_dir, manifest_case_key),
+            "w",
+            encoding="utf8",
+        ) as f:
+            json.dump(MOCK_MANIFEST_CASE[manifest_case_key], f)
+
+        with patch(
+            f"{deadline.__package__}.job_attachments.api.attachment.download_files_from_manifests",
+            return_value=DownloadSummaryStatistics(),
+        ) as mock_download_files_from_manifests:
+            attachment_api.attachment_download(
+                manifests=[os.path.join(temp_assets_dir, manifest_case_key)],
+                s3_root_path="bucket/assetRoot",
+                boto3_session=session_mock,
+            )
+
+            mock_download_files_from_manifests.assert_called_once_with(
+                s3_bucket="bucket",
+                manifests_by_root={
+                    f"{os.getcwd()}/{manifest_case_key}": decode_manifest(
+                        json.dumps(MOCK_MANIFEST_CASE[manifest_case_key])
+                    ),
+                },
+                cas_prefix="assetRoot/Data",
+                session=session_mock,
+            )
+
+
+def test_attachment_download_multiple_to_current(temp_assets_dir):
+    with patch.object(api._session, "get_boto3_session") as session_mock:
+        expected_merged: Dict[str, BaseAssetManifest] = dict()
+
+        for manifest_case_key in MOCK_MANIFEST_CASE.keys():
+            expected_merged[f"{os.getcwd()}/{manifest_case_key}"] = decode_manifest(
+                json.dumps(MOCK_MANIFEST_CASE[manifest_case_key])
+            )
+            with open(
+                os.path.join(temp_assets_dir, manifest_case_key),
+                "w",
+                encoding="utf8",
+            ) as f:
+                json.dump(MOCK_MANIFEST_CASE[manifest_case_key], f)
+
+        with patch(
+            f"{deadline.__package__}.job_attachments.api.attachment.download_files_from_manifests",
+            return_value=DownloadSummaryStatistics(),
+        ) as mock_download_files_from_manifests:
+            attachment_api.attachment_download(
+                manifests=[os.path.join(temp_assets_dir, key) for key in MOCK_MANIFEST_CASE.keys()],
+                s3_root_path="bucket/assetRoot",
+                boto3_session=session_mock,
+            )
+
+            mock_download_files_from_manifests.assert_called_once_with(
+                s3_bucket="bucket",
+                manifests_by_root=expected_merged,
+                cas_prefix="assetRoot/Data",
+                session=session_mock,
+            )
+
+
+def test_attachment_download_invalid_input_manifests(fresh_deadline_config):
+    with patch.object(api._session, "get_boto3_session") as session_mock:
```

Comment:
I see we are patching this `get_boto3_session` quite a bit in this file. Should we make it a `@pytest.fixture(autouse=True)`?

Reply:
Good call, will add it to next PR. It will be used more for other attachment APIs.